SpotifyAPI/SpotifyAPI.ino

```diff
@@ -0,0 +1,233 @@
+                SSLClient.println(CONTENT_TYPE + "application/x-www-form-urlencoded");
+                SSLClient.println(CONTENT_LENGTH + postData.length());
+                SSLClient.println(CONNECTION_CLOSE);
+                SSLClient.println();
+                SSLClient.println(postData);
+            }
+        }
+
+       // Album Art Methods
+       void getAlbumArtURL() {
+           SSLClient.stop();
+           if(SSLClient.connectSSL(spotifyAPI.c_str(), CONNECTION_PORT)) {
+               SSLClient.println(GET + " /v1/me/player/currently-playing " + HTTP_VERSION);
+               SSLClient.println(HOST + spotifyAPI);
+               SSLClient.println(AUTHORIZATION + "Bearer " + oauthToken);
+               SSLClient.println(CONTENT_TYPE + "application/json");
+               SSLClient.println(CONNECTION_CLOSE);
+               SSLClient.println();
+           }
+       }
+
+       void getAlbumArt() {
+           SSLClient.stop();
+           const int beginningIndex = albumArtURL.indexOf("//") + 2;
+           String imageApi = albumArtURL.substring(beginningIndex, albumArtURL.indexOf("/", beginningIndex));
+           String imageExtension = albumArtURL.substring(albumArtURL.indexOf("/", beginningIndex));
+
+           if(SSLClient.connectSSL(imageApi.c_str(), CONNECTION_PORT)) {
+               SSLClient.println(GET + imageExtension + " " + HTTP_VERSION);
+               SSLClient.println(HOST + imageApi);
+               SSLClient.println(AUTHORIZATION + "Bearer " + oauthToken);
+               SSLClient.println(CONNECTION_CLOSE);
+               SSLClient.println();
+           }
+       }
+
+       // Checking methods
+       bool checkHTTPStatus() {
+           char status[32] = {0};
+           SSLClient.readBytesUntil('\r', status, sizeof(status));
+           if (strcmp(status, "HTTP/1.1 200 OK") != 0) {
+               Serial.print("Unexpected response: ");
+               Serial.println(status);
+               SSLClient.stop();
+               return false;
+           }
+           return true;
+       }
+
+       bool skipHTTPHeaders() {
+           char endOfHeaders[] = "\r\n\r\n";
+           if (!SSLClient.find(endOfHeaders)) {
+               Serial.println("Invalid response");
+               SSLClient.stop();
+               return false;
+           }
+       }
+
+    
+    WiFiSSLClient SSLClient;
+
+    String oauthToken;
+    long oauthExpiryTime = 0;
+
+    long nextAlbumArtRetrievalTime = 0;
+    long nextDisplayUpdate = 0;
+    String albumArtURL;
+
+    // Wifi Constants
+    const uint16_t CONNECTION_PORT = 443;
+    String GET = "GET";
+    String POST = "POST";
+    String HOST = "Host: ";
+    String AUTHORIZATION = "Authorization: ";
+    String HTTP_VERSION = "HTTP/1.1";
+    String CONTENT_TYPE = "Content-Type: ";
+    String CONTENT_LENGTH = "Content-Length: ";
+    String CONNECTION_CLOSE = "Connection: close";
+};
+
+class WiFiUtilities {
+
```

Comment:
I like this, but also I feel like the WiFi object should be a member of this. But this brings into question the whole class ownership policy and whether is is ok to pass member vars by reference...

Reply:
I had that same thought. I wasn't entirely sure